onnx/test/cpp/data_propagation_test.cc

```diff
@@ -26,8 +26,8 @@ inline bool CompareShape(
 
   for (int i = 0; i < inferredShape.dim_size(); i++) {
     EXPECT_TRUE(
-        (inferredShape.dim(i).has_dim_value() && expectedShape.dim(i).has_dim_value()) ||
-        (inferredShape.dim(i).has_dim_param() && expectedShape.dim(i).has_dim_param()))
+        (inferredShape.dim(i).has_dim_value() == expectedShape.dim(i).has_dim_value()) &&
+        (inferredShape.dim(i).has_dim_param() == expectedShape.dim(i).has_dim_param()))
```

Comment:
Update here to test with "?" shape (which means no value and no param because it is not computable) -- if the dim does not have value or parameter, it is still a valid dim.